core/controlplane/config/templates/cloud-config-controller

```diff
@@ -1,3 +1,21 @@
+{{ define "instance" -}}
+{{- $S3URI := self.Parts.s3.Asset.URL -}}
```

Comment:
I see, that is default template behaviour where missing map key produces "nothing": https://golang.org/pkg/text/template/#Template.Option

It is more of a case, Cluster objects are produced in a not fully usable state and certain methods must be called in certain order before other methods can be called at all :)

I'll move `Cluster.Assets()` insdie `Cluster` constructor where it belongs to, so that when `Cluster` object is initialized and is ready to be used